### ISSUE TITLE
Outlook: Addins failing when a meeting appointment window is open on some Windows desktops

### ISSUE THREAD

**Issue by garysharp** [OfficeDev/office-js#2055]
> _Edit:_ Originally we thought this was just affecting _OnSend_ handler addins (as presented in this initial post). The scope of this issue broadened quite significantly throughout troubleshooting and diagnosis. All (office-js) addins were failing. The issue was resolved by the Outlook team centrally.

Some users are unable to send (any) meeting appointments. After some time, Outlook reports:

> {Addin name} couldn't complete. Please try sending this message later.

This is limited to meeting appointments on PC Desktop Outlook. Users have no problems sending regular email. MSEdgeWebView2 is not fetching any add-in code. Breakpoints are never hit when debugging. Issue appears to be profile-scoped - a different user on the same device may not have the problem. Our current workaround is to have users send appointments via the web client (any browser).

## Environment
* Platform: some PC desktop only
* Host: Outlook
* Office version number: 2107 (Build 14228.20250 Click-to-Run)
* Operating System: Windows 10 (various versions 1903 through 21H1)
* Browser: N/A
* WebView2: 92.0.902.73

## Context
We have a very simple add-in which uses the OnSend event to perform basic validation of emails before sending them. This has been deployed to over 1,000 users for over 6 months. In troubleshooting we removed our Addin from the equation and attempted to deploy the [sample add-in](https://github.com/OfficeDev/Outlook-Add-in-On-Send). The sample produced the same behavior - sending emails was fine; meeting appointments failed.

## Expected behavior
For add-in code to run consistently across devices/platforms. We have no problem on most desktops or using Outlook on the web (this is our current workaround).

## Current behavior
The first report came in from a user on 2021-08-06. When sending regular emails all is fine, however when sending appointments (which also triggers the OnSend event) Outlook eventually reports:

> {Addin name} couldn't complete. Please try sending this message later.

Sometimes this message appears after 25-30 seconds, other times not until 5 minutes. For affected users this behavior is consistent for all appointments.

## Steps to reproduce
1. Install Add-in (live example below)
2. Create a meeting appointment
3. Add a title and a recipient to the appointment
4. Send the appointment

For affected users, the behavior is as described above

## Link to live example(s)
To narrow down the problem we deployed the sample add-in and have the same problem.
Fork of sample: https://github.com/garysharp/Outlook-Add-in-On-Send ([minor changes](https://github.com/garysharp/Outlook-Add-in-On-Send/commit/0c54f011cc0b2e7c0cd1d818f7eb5022ba03bab3))
Published at: https://onsendaddin.z26.web.core.windows.net/Contoso%20Message%20Body%20Checker.xml

## Troubleshooting
We cleared all caches, including deleting the contents of the [Wef directory](https://docs.microsoft.com/en-us/office/dev/add-ins/testing/troubleshoot-development-errors).
We used [Fiddler](https://www.telerik.com/fiddler) to inspect the MSEdgeWebView2 requests as the add-in was triggered. For affected users, only the initial html page was fetched - none of the JavaScript files were fetched. There was no evidence that the page itself was loaded. When attempted to debug (using Visual Studio or VS Code) breakpoints are never hit.
The MSEdgeWebView2 processes quickly closed after several seconds, yet Outlook continued to display the following message before timing out after 30 seconds or 5 minutes:

> (spinner) {Addin name} is working on your request.

Greatly appreciate any assistance that can be provided.

**Comment by exextoc** [comment]
When this happens to users, it's a 100% repro for them? Can they still open other add-ins in Taskpane? 

This error usually happens when either the add-in cannot be loaded from the webpage (i.e. the connection to the webserver hosting the add-in is down), or the add-in does not call event.completed in 5 minutes. If it is failing in less than 5 minutes, I would expect trouble reaching the website as the cause, but I would also expect it to succeed for that user later. (it doesn't sound like this is the case)

Sending Email and sending appointments should run the add-in the same way, which is curious that this error never happens on email, and leads me to suspect this may be a new issue that we don't know about. (we have not been able to repro it on our end)

Does uninstalling WebView2 affect this in any way? Does recreating the profile for an affected user affect anything? Or if an affected user uses a different machine?

Any additional information that you can provide about the users experiencing the issue will help us narrow down the issue. (i.e. is there something common/unique among affected users).

**Comment by garysharp** [comment]
Thanks for this @exextoc.

> When this happens to users, it's a 100% repro for them?

Yes. I can say that this is happening 100% of the time for affected users.

> This error usually happens when either the add-in cannot be loaded from the webpage ...

We were able to reproduce when hosting the add-in via localhost. I can also confirm (by using Fiddler) that all (attempted) requests to the server are successful and returning the expected payload. In our production environment the add-in is hosted on Azure Blob Storage (static website) behind an Azure CDN. There is no dynamic server-side code.

>  ... or the add-in does not call event.completed in 5 minutes.

We encountered the 5 minute timeout of `event.completed` during original development with extended debugging sessions. This would explain the 5 minute failure (though not the 25-30 second failures). This would also be expected if the JavaScript code is never fetched (or evaluated) as we believe to be the case here. In our testing, calling `event.completed` from within an inline script tag of the add-in html handler still fails. Explicit calls to `notificationMessages.addAsync` also does not result in any messages shown.

> If it is failing in less than 5 minutes, I would expect trouble reaching the website as the cause, but I would also expect it to succeed for that user later.

This add-in is deployed to over 1,000 users across Australia with many working from home (accessing the CDN endpoint via their own ISP not any corporate WAN). It is affected users across different hardware types, geographical locations, job positions. At this time we have not been able to narrow down any commonality between users.

> Or if an affected user uses a different machine?

If an affected user uses a different machine it is likely to work (has in all of our attempts). If another user uses the affected users machine (building a new profile) it is likely to work (has in all of our attempts). This leads us to believe it is profile-bound.

I acknowledge that I have not addressed all the questions/scenarios you pose. We are actively following up on these and will provide concrete answers ASAP.

**Comment by neville-jones** [comment]
We are seeing reports of similar problems with a few users, but not all. Seems limited to Outlook Current Channel (or Current Preview) on Windows desktops; older Outlooks don't show the problem.

Interestingly, for users affected by the problem, if they attempt to start the standard Microsoft "My Templates" add-in in the meeting invite then it fails to load, coming up with a yellow warning and suggesting to restart Outlook.

Some more information from one of our users:

> So I uninstalled Office 365, and did a clean install of Office 365 v16.0.13029.20344 and all add-ins worked correctly as expected. Then updating to v16.0.14228.20250 broke all the portal add-ins again. (in a meeting invite)
> I can use the ‘room finder’ function in the Calendar without any issues.
> And deleting and recreating the mail profile did not resolve the issue for me.

**Comment by garysharp** [comment]
The team has coordinated a series of tests based on your feedback @exextoc.

Firstly, we simplified the add-in as much as possible. All tests were performed with our production add-in, the [MS sample add-in](https://github.com/OfficeDev/Outlook-Add-in-On-Send), and a [minimal and minimal-inline add-in](https://github.com/garysharp/Outlook-Add-in-On-Send/commit/2fc73b8fc2edec0caa2dbae7449d74feb807b340). Results were consistent across all tests for all add-ins.

We confirmed that if we re-created the user profile the problem was likely fixed (it was in each of our tests).

We confirmed that if we uninstalled WebView2 the problem didn't occur. After re-installing WebView2 the problem re-appeared.

We are continuing to work the problem here, running into some dead-ends though.

@neville-jones - I'll try and coordinate some testing of what you've described in our environment.

Please let us know if we can perform additional testing.

**Comment by exextoc** [comment]
> We confirmed that if we uninstalled WebView2 the problem didn't occur. After re-installing WebView2 the problem re-appeared.

This is an interesting note. And probably why we are only seeing this on recent builds. I am going to try and foward this issue to the appropriate dev owners and see if they have any ideas. It is likely some additional logging will need to take place. 

Usually the first step is to grab ULS Logging: 

https://docs.microsoft.com/en-us/office365/troubleshoot/diagnostic-logs/collect-office-diagnostic-logs 

(hopefully you can grab that info) 

> We confirmed that if we re-created the user profile the problem was likely fixed (it was in each of our tests).

I realize I was unclear about this before. Did you mean the **Windows** profile, or the **Outlook** profile?

**Comment by neville-jones** [comment]
I note that uninstalling WebView2 forces Outlook 2107 (Build 14228.20250 Click-to-Run) to use the original WebView (not based on Edge Chromium). Will ask our affected user to try this approach and see if it 'fixes' the problem. I speculate that Outlook might restore WebView2 in the background so might only be a temporary workaround.

**Comment by garysharp** [comment]
We've performed some additional testing and ULS logs have been emailed to @croicu - thank you.

We _had_ tested recreating the **Windows** user profile. I can confirm that recreating the **Outlook** profile has no effect - the problem still occurs.

It does seem to be a broader issue than just OnSend handlers.

As @neville-jones noted, our affected users are also not able to load "My Templates":
![MeetingMyTemplatesError](https://user-images.githubusercontent.com/1687386/130395629-01b33e1d-1df8-44e1-ba54-158d7e0ff4ed.png)
Attempting to use the office-js Addin for Zoom also fails (the COM addin _does_ work).

Users also identified another quirk. If a meeting window was open, they were unable to send regular emails. Reproduction steps:
1. Open a "New Meeting" window
2. Open a "New Email" window
3. Enter a recipient and attempt to send the email.
The behavior is then the same as for meetings, the email fails to send after about 30 seconds.

**Comment by SylvaD-MS** [comment]
I have looked at the ULS logs which were provided and I can find the error above.  I can also repro the issue; if I choose a different locale for Office without closing all the Office Apps especially if a Web Add-in is already activated.  No new Web Add-in will be activated until the previous Office App has been closed and the existing WebView2 process (msedgewebview2.exe)  has been terminated.

Unfortunately, this isn't a supported scenario (running Office side by side with different locales) and all the Office Apps must be closed whenever the Office UI locale is changed.  This is the current design when using WebView2.

**Comment by garysharp** [comment]
Thanks for stepping in @SylvaD-MS. 

This doesn't appear the cause of our problem. I can confirm that our affected users _are not_ changing their locale settings. 

(more details and screenshots have been emailed)

**Comment by neville-jones** [comment]
Thanks @SylvaD-MS but like @garysharp our users have not been changing their locale settings. Our users are largely based in Australia and New Zealand; Gary's are in Australia. 

Maybe it is something to do with the users have a locale setting in Office or Windows of other than en-US?

**Comment by neville-jones** [comment]
We have more customers reporting this problem occurring for some of their users.

@SylvaD-MS - we believe the users do **not** have Office running side by side with different locales so how do we go about having Microsoft continue investigation into this issue?

**Comment by SylvaD-MS** [comment]
As discussed, the issue is a known issue which has been fixed but unfortunately not 100% deployed worldwide.  It is somewhat specific to Outlook (but could also affect other Office Apps depending on the order of the Web Add-ins activation).  If you are currently reproing the issue then a mitigation is to have matching **Office Display language** and **Office Authoring Languages and proofing** settings under Outlook options.  You'll need to close all the Office Applications in order to mitigate the issue.

**Comment by SylvaD-MS** [comment]
Thanks for reporting this issue and providing us detail information.  This turned out to be a variant of a known issue and a mitigation is being deployed to prevent it.  In parallel, a fix is getting implemented to avoid this issue to occur in the future.

**Comment by neville-jones** [comment]
One of our affected users upgraded to Outlook on Windows desktop to Version 2108 Build 14326.20238 and it did not rectify the problem - all Outlook web add-ins not functioning in meeting invite. Some users of the organisation had experienced intermittent issues with web add-ins when sending email messages and this Outlook release seems to helped alleviate the problems for them.

The affected user has reported that the issue does indeed seem to be bound to the desktop user profile:

> Just to let you know, further testing on our end is pointing to the issue being local profile related.
> Logging into my machine (which has the issue) with a different user account I cannot replicate the problem.
> Logging into a different computer (which has the latest webview2 and Office update) with my user account I was unable to reproduce the issue either.

**Comment by neville-jones** [comment]
The affected user mentioned above has fixed this issue by changing the following setting in Outlook and restarting Outlook:

![image001 (1)](https://user-images.githubusercontent.com/62081302/131935132-5e8e5251-3395-44e6-99c4-1d36dcfe8a8d.png)

The user is not on a multi-display workstation, just a laptop with single display but the change rectified the problems with web add-ins not working in meeting invite.

**Comment by garysharp** [comment]
## Resolution
It seems that some new functionality in Outlook was being gradually rolled out. The team were able to identify the cause and revert the roll out centrally. This should have come into effect on or about Wednesday 2021-09-01 3:30am UTC (1:30pm AEST, 3:30pm NZST). Users restarted Outlook for the change to be applied.
Thanks for providing support @SylvaD-MS, @croicu and team.
